src/WopiTokenManager.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\collabora_online;
+
+use Drupal\collabora_online\Exception\CollaboraNotAvailableException;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\key\KeyRepositoryInterface;
+use Firebase\JWT\JWT;
+use Firebase\JWT\Key;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+/**
+ * Service to manage the access token to be used by the WOPI client.
+ */
+class WopiTokenManager {
```

Comment:
The JWT part is encapsulated into this service.
From the outside we don't care about it, we only care that it produces a key which it can later decode.
And this token is used for the WOPI requests. This is why I called it that.

Reply:
So at least the interface does not need to have JWT in the name.

Reply:
But you are defining a service name based on how it's used rather than what it does. That's why I suggested Collabora, what about CollaboraJwtTokenManager and CollaboraTokenManagerInterface? We could remove jwt from class name if does not convince you.